config/manager/manager.yaml

```diff
@@ -77,13 +77,18 @@ spec:
             seccompProfile:
               type: RuntimeDefault
           volumeMounts:
+          - mountPath: /tmp
+            name: temp-vol
+            readOnly: false
```

Comment:
we need to add this volume to the helm chart too

Reply:
Isn't it better to store the keytab in some folder other than tmp? I don't think it's a good practice to share tmp between pods.

Reply:
What do you mean? We are not sharing `/tmp` between pods, each pod will create its own file each time that the scaler is built. But you have discovered me another potential issue, orphan files in `/tmp`. We need to delete the file on scaler closing

Reply:
@novicr , could you add the code to delete the file on scaler close function? We should track if the file's been created and if it's been created, we should delete it on close to not keep orphan files. On long-running instances, the pod could have a lot of files (there are multiple reasons that could regenerate the scaler, generating new files)

Reply:
> We are not sharing `/tmp` between pods, each pod will create its own file each time that the scaler is built.

Ahh ok. My concern was that tmp generally implies that it's ephemeral.

It makes sense for the keytab to be in tmp then.

Reply:
I am not 100% sure we should store the keytab there, isn't it a possible security problem? Maybe an in-memory storage?

Reply:
> we need to add this volume to the helm chart too

Will do once we agree on the content.  There are clearly some legitimate changes pending here.

Reply:
> I am not 100% sure we should store the keytab there, isn't it a possible security problem? Maybe an in-memory storage?

Agreed.  Adding medium to keep this in memory.

Reply:
> @novicr , could you add the code to delete the file on scaler close function? We should track if the file's been created and if it's been created, we should delete it on close to not keep orphan files. On long-running instances, the pod could have a lot of files (there are multiple reasons that could regenerate the scaler, generating new files)

@JorTurFer , can scaler be closed and reopened in the context of the same Pod?  I did add file removal.

Reply:
> we need to add this volume to the helm chart too

Added a chart pr: https://github.com/kedacore/charts/pull/509

Reply:
> @JorTurFer , can scaler be closed and reopened in the context of the same Pod? I did add file removal.

Yep. Ideally, the scaler is created once and closed only once because it's stored inside a scaler cache, but if there are failures, the scaler is regenerated. For example, if there is any transient error or any error in the upstream, the scaler is regenerated.
In the worst case with the upstream unreachable, the scaler will be opened and close twice per each reconcilliation loop, that's why I asked for removing the file on closing.